packages/core/echo/echo-schema/src/query/query.test.ts

```diff
@@ -254,4 +254,26 @@ testWithAutomerge(() => {
       }
     });
   });
+
+  test('map over refs in query result', async () => {
+    const testBuilder = new TestBuilder();
+    const peer = await testBuilder.createPeer();
+
+    const folder = peer.db.add(new TypedObject({ name: 'folder', objects: [] }));
+    const objects = [
+      new TypedObject({ idx: 0, title: 'Task 0' }),
+      new TypedObject({ idx: 1, title: 'Task 1' }),
+      new TypedObject({ idx: 2, title: 'Task 2' }),
+    ];
+    for (const object of objects) {
+      folder.objects.push(object);
+    }
```

Comment:
inline syntax should work as well
```suggestion
    const folder = peer.db.add(new TypedObject({ name: 'folder', objects: [
      new TypedObject({ idx: 0, title: 'Task 0' }),
      new TypedObject({ idx: 1, title: 'Task 1' }),
      new TypedObject({ idx: 2, title: 'Task 2' }),
    ]
    }));
```

Reply:
It references objects array later in expect